spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTransformConverter.java

```diff
@@ -132,10 +196,12 @@ public static org.apache.spark.sql.connector.expressions.Transform[] toSparkTran
               });
     }
 
-    org.apache.spark.sql.connector.expressions.Transform bucketTransform =
-        toSparkBucketTransform(distribution, sortOrder);
-    if (bucketTransform != null) {
-      sparkTransforms.add(bucketTransform);
+    if (!isBucketTransform.get()) {
+      org.apache.spark.sql.connector.expressions.Transform bucketTransform =
+          toSparkBucketTransform(distribution, sortOrder);
```

Comment:
Does Iceberg use `BucketTransform` to represent distribution and sortOrder information?

Reply:
BucketTransform can simply be understood as a partitioning function that calculates which partition the record should go into based on the value of the specified field, not distribution, not sortOrder.

But distribution can be hash based on BucketTransform expressions such as bucket(2, id).
sortOrder can be sorted based on the expression of BucketTransform.

Reply:
maybe we can use `supportsBucketTransform` to replace `!isBucketTransform.get()`?